### PR TITLE
Display availability

### PR DESCRIPTION
**Pull request by betrevisan**
The commits proposed in this PR allow the user to see the availability of a tutor and select a desired available time slot for a tutoring session. This procedure is handled by the AvailabilityServlet along with the added changes to availability.java. JavaScript and Java Unit tests are included.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/AvailabilityServlet.java [hunk 1]
```diff
@@ -14,14 +14,39 @@
...
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.*; // Can be deleted after hardcoded database is removed
+
 
 @WebServlet("/availability")
 public class AvailabilityServlet extends HttpServlet {
+    private static final int TIME_0800AM = TimeRange.getTimeInMinutes(8, 00);
```
Comment (kashiish) [inline src/main/java/com/google/sps/servlets/AvailabilityServlet.java:34]: I think these would be good to add in the SampleData class

File: src/main/java/com/google/sps/servlets/AvailabilityServlet.java [hunk 2]
```diff
@@ -31,7 +56,27 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        response.setContentType("plain/text");
-        response.getWriter().println("To be implemented");
+        // Get the id of the tutor whose availability will be displayed.
+        String tutorID = Optional.ofNullable(request.getParameter("tutorID")).orElse(null);
```
Comment (kashiish) [inline src/main/java/com/google/sps/servlets/AvailabilityServlet.java:60]: I don't really think it makes sense to use an Optional here if the default is null.

File: src/main/java/com/google/sps/servlets/AvailabilityServlet.java [hunk 3]
```diff
@@ -31,7 +56,27 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+
+        List<TimeRange> timeslots = new ArrayList<TimeRange>();
+
+        for (Tutor tutor : hardcoded) {
+            if (tutorID.equals(tutor.getEmail())) {
```
Comment (kashiish) [inline src/main/java/com/google/sps/servlets/AvailabilityServlet.java:65]:
```suggestion
            if (tutorID.toLowerCase().equals(tutor.getEmail().toLowerCase())) {
```
Make it case insensitive?

File: src/main/webapp/webapptests/spec/SpecAvailability.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+describe("Availability", function() {
+
+    describe("when the user requests to see a tutor's availability", function() {
+        var mockWindow = {location: {href: "availability.html?tutorID=test%40gmail.com", search: "?tutorID=test%40gmail.com"}};
+
+        it("should trigger the fetch function", function() {
+            spyOn(window, "onload").and.callFake(function() {
+                readTutorID(queryString, window);
+            });
+            spyOn(window, 'fetch').and.callThrough();
```
Comment (kashiish) [inline src/main/webapp/webapptests/spec/SpecAvailability.js:24]: I read that callThrough() executes the real implementation. Should we be doing that in tests?

File: src/test/java/com/google/sps/AvailabilityTest.java [hunk 5]
```diff
@@ -14,17 +14,37 @@
...
+    private static final int TIME_0800AM = TimeRange.getTimeInMinutes(8, 00);
+    private static final int TIME_0900AM = TimeRange.getTimeInMinutes(9, 00);
+    private static final int TIME_1000AM = TimeRange.getTimeInMinutes(10, 00);
+    private static final int TIME_1100AM = TimeRange.getTimeInMinutes(11, 00);
+    private static final int TIME_1200AM = TimeRange.getTimeInMinutes(12, 00);
 
     @Test
-    public void noFailTest() {
-        Assert.assertEquals(true, true);
+    public void testConvertToJsonUsingGson() {
```
Comment (kashiish) [inline src/test/java/com/google/sps/AvailabilityTest.java:35]: I think you should maybe add a test for your doPost method and make sure it returns the correct response (like the correct tutor's timeslots for the id parameter).

File: src/main/webapp/webapptests/spec/SpecAvailability.js (hunk 4 above)
Comment (betrevisan) [reply src/main/webapp/webapptests/spec/SpecAvailability.js:24]: In this case, I wanted fetch to run normally. The reason I created the spy was to be able to know in the end if had been successfully executed. I wouldn't be able to do expect(window.fetch).toHaveBeenCalled() if I hadn't spied on fetch.

File: src/main/java/com/google/sps/servlets/AvailabilityServlet.java (hunk 1 above)
Comment (betrevisan) [reply src/main/java/com/google/sps/servlets/AvailabilityServlet.java:34]: I agree, I'll add this as a new commit.

File: src/main/java/com/google/sps/servlets/AvailabilityServlet.java (hunk 2 above)
Comment (betrevisan) [reply src/main/java/com/google/sps/servlets/AvailabilityServlet.java:60]: That makes a lot of sense. I just removed it.

File: src/main/java/com/google/sps/servlets/AvailabilityServlet.java (hunk 3 above)
Comment (betrevisan) [reply src/main/java/com/google/sps/servlets/AvailabilityServlet.java:65]: I agree. I'll submit it as new commit here.

File: src/main/webapp/webapptests/spec/SpecAvailability.js (hunk 4 above)
Comment (kashiish) [reply src/main/webapp/webapptests/spec/SpecAvailability.js:24]: Ok, that makes sense!

File: src/test/java/com/google/sps/AvailabilityTest.java (hunk 5 above)
Comment (betrevisan) [reply src/test/java/com/google/sps/AvailabilityTest.java:66]: Yes, that's important. My newest commit to this PR addresses that test.

Comment (kashiish) [reply src/test/java/com/google/sps/AvailabilityTest.java:66]: Awesome!

**Pull request merged**